pkg/storage/cache.go

```diff
@@ -1,301 +1,41 @@
 package storage
 
 import (
-	"path"
-	"path/filepath"
-	"strings"
-	"time"
-
 	godigest "github.com/opencontainers/go-digest"
-	"go.etcd.io/bbolt"
 
 	"zotregistry.io/zot/errors"
 	zlog "zotregistry.io/zot/pkg/log"
+	"zotregistry.io/zot/pkg/storage/cache"
 )
 
-const (
-	// global bucket.
-	BlobsCache = "blobs"
-	// bucket where we store all blobs from storage(deduped blobs + original blob).
-	DuplicatesBucket = "duplicates"
-	/* bucket where we store only the original/source blob (used by s3 to know which is the blob with content)
-	it should contain only one blob, this is the only place from which we'll get blobs. */
-	OriginalBucket          = "original"
-	DBExtensionName         = ".db"
-	dbCacheLockCheckTimeout = 10 * time.Second
-)
-
-type Cache struct {
-	rootDir     string
-	db          *bbolt.DB
-	log         zlog.Logger
-	useRelPaths bool // weather or not to use relative paths, should be true for filesystem and false for s3
-}
-
-// Blob is a blob record.
-type Blob struct {
-	Path string
-}
-
-func NewCache(rootDir string, name string, useRelPaths bool, log zlog.Logger) *Cache {
-	dbPath := path.Join(rootDir, name+DBExtensionName)
-	dbOpts := &bbolt.Options{
-		Timeout:      dbCacheLockCheckTimeout,
-		FreelistType: bbolt.FreelistArrayType,
-	}
-
-	cacheDB, err := bbolt.Open(dbPath, 0o600, dbOpts) //nolint:gomnd
-	if err != nil {
-		log.Error().Err(err).Str("dbPath", dbPath).Msg("unable to create cache db")
+type Cache interface {
+	// Returns the human-readable "name" of the driver.
+	Name() string
 
-		return nil
-	}
-
-	if err := cacheDB.Update(func(tx *bbolt.Tx) error {
-		if _, err := tx.CreateBucketIfNotExists([]byte(BlobsCache)); err != nil {
-			// this is a serious failure
-			log.Error().Err(err).Str("dbPath", dbPath).Msg("unable to create a root bucket")
+	// Retrieves the blob matching provided digest.
+	GetBlob(digest godigest.Digest) (string, error)
 
-			return err
-		}
+	// Uploads blob to cachedb.
+	PutBlob(digest godigest.Digest, path string) error
 
-		return nil
-	}); err != nil {
-		// something went wrong
-		log.Error().Err(err).Msg("unable to create a cache")
+	// Check if blob exists in cachedb.
+	HasBlob(digest godigest.Digest, path string) bool
 
-		return nil
-	}
-
-	return &Cache{rootDir: rootDir, db: cacheDB, useRelPaths: useRelPaths, log: log}
+	// Delete a blob from the cachedb.
+	DeleteBlob(digest godigest.Digest, path string) error
 }
 
-func (c *Cache) PutBlob(digest godigest.Digest, path string) error {
-	if path == "" {
-		c.log.Error().Err(errors.ErrEmptyValue).Str("digest", digest.String()).Msg("empty path provided")
-
-		return errors.ErrEmptyValue
-	}
+func Create(dbtype string, parameters interface{}, log zlog.Logger) (Cache, error) {
+	switch dbtype {
+	case "boltdb":
+		{
+			boltDriverFactory := cache.BoltDBDriverFactory{}
```

Comment:
Just make this boltdb.Create() <- public method, remove all Factory references

Reply:
Modified as requested